File: ProjetoPadraoDotnetCore/Aplication/Models/Request/Projeto/ProjetoRequest.cs
using Infraestrutura.Enum;

namespace Aplication.Models.Request.Projeto;

public class ProjetoRequest
{
    public int? IdProjeto { get; set; }
    public int IdUsuarioCadastro { get; set; }
    public string? Titulo { get; set; }
    public string? Foto { get; set; }
    public DateTime DataInicio { get; set; }
    public DateTime DataFim { get; set; }
    public string? Descricao { get; set; }
    public bool EmailProjetoAtrasado { get; set; }
    public bool PortalProjetoAtrasado { get; set; }
    public bool EmailTarefaAtrasada { get; set; }
    public bool PortalTarefaAtrasada { get; set; }
    public bool AlteracaoStatusProjetoNotificar { get; set; }
    public bool AlteracaoTarefasProjetoNotificar { get; set; }

    public List<AtividadeRequest>? Atividade { get; set; }
    public List<TarefaUsuarioRequest>? Tarefa { get; set; }

    public class AtividadeRequest
    {
        public int? IdAtividade { get; set; }
        public string? Atividade { get; set; }
        public DateTime DataInicial { get; set; }
        public DateTime DataFim { get; set; }
        public EStatusAtividade StatusAtividade { get; set; }
        public List<TarefaRequest>? ListTarefas { get; set; }
    }

    public class TarefaUsuarioRequest
    {
        public int ResponsavelId { get; set; }
        public List<TarefaAtividadeRequest>? Tarefa { get; set; }
    }

    public class TarefaRequest
    {
        public int? IdTarefa { get; set; }
        public string? Descricao { get; set; }
        public string? DescricaoTarefa { get; set; }
        public EPrioridadeTarefa Prioridade { get; set; }
        public List<string>? LTagsTarefa { get; set; }
    }

    public class TarefaAtividadeRequest
    {
        public int? IdTarefa { get; set; }
        public string? Tarefa { get; set; }
        public string? Atividade { get; set; }
    }
}

File: ProjetoPadraoAngular/src/app/objects/Projeto/ProjetoRequest.ts
import { EStatusAtividade } from "src/app/enums/EStatusAtividade";
import { Tarefa } from "../Tarefa/Tarefas";

export interface ProjetoRequest
{
    IdProjeto: number | undefined,
    Titulo: string,
    DataInicio: string,
    DataFim: string,
    Descricao: string | undefined,
    EmailProjetoAtrasado: boolean,
    PortalProjetoAtrasado: boolean,
    EmailTarefaAtrasada: boolean,
    PortalTarefaAtrasada: boolean,
    AlteracaoStatusProjetoNotificar: boolean,
    AlteracaoTarefasProjetoNotificar: boolean,
    Atividade: Array<AtividadeRequest>, 
    IdUsuarioCadastro: number,
    Tarefa: Array<any>,
    Foto: string | undefined
}

export interface AtividadeRequest
{
    IdAtividade: number | undefined,
    Atividade: string,
    DataInicial: Date | undefined,
    DataFim: Date | undefined,
    ListTarefas: Array<Tarefa>
    StatusAtividade: EStatusAtividade
}
